### PR TITLE
[SPARK-36097][CORE] Grouping exception in core/scheduler

### PR DESCRIPTION
**Pull request by dgd-contributor**
### What changes were proposed in this pull request?
This PR group exception messages in core/src/main/scala/org/apache/spark/scheduler

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  def nonExistentAccumulatorError(id: Long): Throwable = {
+    new SparkException(s"attempted to access non-existent accumulator $id")
+  }
+
+  def sendResubmittedTaskStatusForShuffleMapStagesOnlyError(): Throwable = {
+    new SparkException("TaskSetManagers should only send Resubmitted task " +
+      "statuses for tasks in ShuffleMapStages.")
+  }
+
+  def emptyEventQueueTimeoutError(timeoutMillis: Long): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:68]: nonEmptyEventQueueAfterTimeoutError

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    new TimeoutException(s"The event queue is not empty after $timeoutMillis ms.")
+  }
+
+  def durationOperationUnsupportedOnUnfinishedTaskError(): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:72]: durationCalledOnUnfinishedTaskError

File: core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala [hunk 3]
```diff
@@ -916,7 +918,7 @@ private[spark] class TaskSchedulerImpl(
         // No task sets are active but we still got an error. Just exit since this
         // must mean the error is during registration.
         // It might be good to do something smarter here in the future.
-        throw new SparkException(s"Exiting due to error from cluster scheduler: $message")
+        throw SparkCoreErrors.clusterSchedulerError(message)
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala:921]: The error message seems changed here.

File: core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+  def markExecutorAsFailedError(errorMsg: String): Throwable = {
+    new SparkException(errorMsg)
+  }
+
+  def clusterSchedulerError(message: String): Throwable = {
+    new SparkException(message)
+  }
+
+  def serializationTaskError(e: Throwable): Throwable = {
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala:94]: failToSerializeTaskError

**Review by allisonwang-db — COMMENTED** [review]
Left a few comments. Please also resolve the merge conflict.

**Comment by dgd-contributor** [comment]
@allisonwang-db I've rebase and resolve all of your suggestions. Can you re-review.

**Review by allisonwang-db — APPROVED** [review]
Looks good.

**Comment by dgd-contributor** [comment]
cc @cloud-fan

**Comment by mridulm** [comment]
@cloud-fan There are a few of these PR's, wanted to surface this [comment](https://github.com/apache/spark/pull/33555#issuecomment-888457574) - to make sure are interfaces are tighter.

I dont want to post this in all PR's :-)

**Comment by dgd-contributor** [comment]
> @cloud-fan There are a few of these PR's, wanted to surface this [comment](https://github.com/apache/spark/pull/33555#issuecomment-888457574) - to make sure are interfaces are tighter.
> 
> I dont want to post this in all PR's :-)

cc @allisonwang-db as well

**Comment by dgd-contributor** [comment]
cc @cloud-fan

**Comment by dgd-contributor** [comment]
cc @cloud-fan I've resolve the conflict file, can you review this

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**